src/init_interp/test/initInterpMindist_test/test_set_mindist.pf

```diff
@@ -724,6 +735,166 @@ contains
     end associate
   end subroutine newveg_usesBaresoil
 
+  @Test
+  subroutine TBDurban_usesHDurban(this)
+    ! If there's a new urban TBD type, this should take inputs from the closest
+    ! HD type, if fill_missing_urban_with_HD = .true and fill_missing_with_natveg = .false.
+    !
+    class(TestSetMindist), intent(inout) :: this
+    type(subgrid_type) :: subgridi, subgrido
+    real(r8), parameter :: my_lat = 31._r8
+    real(r8), parameter :: my_lon = 41._r8
+    integer :: i
+    integer :: mindist_index(1)
+
+    associate( &
+         ilun_urban_TBD => subgrid_special_indices%ilun_urban_TBD, &
+         ilun_urban_HD => subgrid_special_indices%ilun_urban_HD, &
+         ilun_urban_MD => subgrid_special_indices%ilun_urban_MD &
+         )
+
+    call setup_landunit_ncols(ltype=ilun_urban_TBD, &
+         ctypes=[71,72,73,74,75], &
+         cweights=[0.6_r8,0.1_r8,0.1_r8,0.1_r8,0.1_r8], &
+         ptype=0)
+
+    call create_subgrid_info( &
+         subgrid_info = subgrido, &
+         npts = 1, &
+         beg = 1, &
+         name = 'landunit', &
+         ltype = [ilun_urban_TBD], &
+         lat = [my_lat], &
+         lon = [my_lon])
+
+    ! Input points differ in landunit type
+    call create_subgrid_info( &
+         subgrid_info = subgridi, &
+         npts = 2, &
+         name = 'landunit', &
+         ltype = [ilun_urban_MD, ilun_urban_HD], &
+         lat = [(my_lat, i=1,2)], &
+         lon = [(my_lon, i=1,2)])
+
+    call wrap_set_mindist(subgridi, subgrido, mindist_index, &
+         fill_missing_urban_with_HD = .true., &
+         fill_missing_with_natveg = .false.)
+
```

Comment:
Meeting with @olyson we noted that it's really subtle that we are checking that the index compared here is of the array that was sent in in the call on 775. This should be made more obvious, or commented on or something.